go/node/cmd/main.go

```diff
@@ -0,0 +1,44 @@
+package main
+
+import (
+	"github.com/obscuronet/go-obscuro/go/node"
+)
+
+func main() {
+	cliConfig := ParseConfigCLI()
+	// todo: allow for multiple operation (start, stop, status)
+
+	nodeCfg := node.NewNodeConfig(
+		node.WithNodeName(cliConfig.nodeName),
```

Comment:
I love this pattern but feel like it mostly makes sense in a more DSL-style scenario. Especially one where most of the values will be defaulted and the caller will be just overriding one or two. Is that how you see this mostly being used?

Reply:
I see this being used in multiple instances (like the testnet) that's why it's using functional options.
As for the defaults I'd preferably want to avoid them, they require implicit knowledge which I try to avoid.
We can, however, build that pattern on top of this one, I've used something similar in the past.

Something like

```
nodeCfg := node.NewNodeConfig( nodetypes.DefaultSequencer(), node.WithEnclaveImage(...) )
```

Where the ` nodetypes.DefaultSequencer()` preloads a bunch of defaults. But, these are defaults of the ` nodetypes.DefaultSequencer()` not of the `node`. If that makes sense ?

Reply:
Cool cool, yeah that's a nice pattern